### PR TITLE
Reorder Asier's commits in order to deploy CRI fixes

### PR DESCRIPTION
**Pull request by vchske**
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/1905)

### CODE REVIEW

File: src/Common/CosmosClient.test.ts [hunk 1]
```diff
@@ -28,19 +28,6 @@ describe("tokenProvider", () => {
   afterEach(() => {
     jest.restoreAllMocks();
   });
-
```
Comment (languy) [inline src/Common/CosmosClient.test.ts:31]: FYI, the `describe("tokenProvider", () => {})` doesn't have any tests anymore and should be removed or the tests should be fixed if the path is still used?

File: src/Common/CosmosClient.ts [hunk 2]
```diff
@@ -17,7 +17,16 @@ const _global = typeof self === "undefined" ? window : self;
...
 
-  if (userContext.features.enableAadDataPlane && userContext.aadToken) {
+  const aadDataPlaneFeatureEnabled =
+    userContext.features.enableAadDataPlane && userContext.databaseAccount.properties.disableLocalAuth;
+  const dataPlaneRBACOptionEnabled = userContext.dataPlaneRbacEnabled && userContext.apiType === "SQL";
+  if (aadDataPlaneFeatureEnabled || (!userContext.features.enableAadDataPlane && dataPlaneRBACOptionEnabled)) {
+    if (!userContext.aadToken) {
+      logConsoleError(
+        `AAD token does not exist. Please use "Login for Entra ID" prior to performing Entra ID RBAC operations`,
+      );
```
Comment (languy) [inline src/Common/CosmosClient.ts:27]: Add send some telemetry?

File: src/Explorer/Explorer.tsx [hunk 3]
```diff
@@ -251,8 +254,44 @@ export default class Explorer {
...
+        useDataPlaneRbac.setState({ aadTokenUpdated: true });
+      } catch (error) {
+        if (error instanceof msal.AuthError && error.errorCode === msal.BrowserAuthErrorMessage.popUpWindowError.code) {
+          logConsoleError(
+            "We were unable to establish authorization for this account, due to pop-ups being disabled in the browser.\nPlease enable pop-ups for this site and try again",
+          );
+        } else {
+          const errorJson = JSON.stringify(error);
+          logConsoleError(
+            `Failed to perform authorization for this account, due to the following error: \n${errorJson}`,
```
Comment (languy) [inline src/Explorer/Explorer.tsx:290]: Send telemetry?

File: src/hooks/useKnockoutExplorer.ts [hunk 4]
```diff
@@ -390,6 +423,19 @@ function configureEmulator(): Explorer {
...
+async function fetchAndUpdateKeys(subscriptionId: string, resourceGroup: string, account: string) {
+  try {
+    const keys = await listKeys(subscriptionId, resourceGroup, account);
+
+    updateUserContext({
+      masterKey: keys.primaryMasterKey,
+    });
+  } catch (error) {
+    console.error("Error during fetching keys or updating user context:", error);
+    throw error;
```
Comment (languy) [inline src/hooks/useKnockoutExplorer.ts:435]: Send telemetry?

File: src/hooks/useKnockoutExplorer.ts [hunk 5]
```diff
@@ -443,6 +489,39 @@ async function configurePortal(): Promise<Explorer> {
...
+              if (!dataPlaneRbacEnabled) {
+                await fetchAndUpdateKeys(subscriptionId, resourceGroup, account.name);
+              }
+
+              updateUserContext({ dataPlaneRbacEnabled });
+              useDataPlaneRbac.setState({ dataPlaneRbacEnabled: dataPlaneRbacEnabled });
+            }
+          } else {
+            await fetchAndUpdateKeys(subscriptionId, resourceGroup, account.name);
+          }
```
Comment (languy) [inline src/hooks/useKnockoutExplorer.ts:523]: How different is this logic from line 275-310? Is there code duplication we should avoid?

**Review by languy — APPROVED** [review]

File: src/hooks/useKnockoutExplorer.ts (hunk 5 above)
Comment (sindhuba) [reply src/hooks/useKnockoutExplorer.ts:523]: I'm addressing this one and other comments related to RBAC in a separate PR. Thanks .

**Pull request merged**